src/librustc_mir/transform/add_yk_swt_calls.rs

```diff
@@ -131,51 +131,54 @@ impl MirPass for AddYkSWTCalls {
     }
 }
 
-/// Given a `MirSource`, decides if we should annotate the correpsonding MIR.
-fn should_annotate(tcx: TyCtxt<'a, 'tcx, 'tcx>, src: MirSource) -> bool {
-    // Never annotate any MIR-like thing marked `#[no_trace]` or `#[naked]`. The trace record and
-    // wrapper are also marked `#[no_trace]` to prevent infinite recursion.
+/// Given a `MirSource`, decides if it is possible for us to trace (and thus whether we should
+/// transform) the MIR. Returns `true` if we cannot trace, otherwise `false`.
+fn is_untraceable(tcx: TyCtxt<'a, 'tcx, 'tcx>, src: MirSource) -> bool {
```

Comment:
Hrm, something is only untraceable if it's not possible to apply the transform. i.e. "untraceable" here is defined in terms of the transform. The concepts are tightly coupled. Unless you feel strongly, I'd rather leave it in.

Reply:
You said above that you'll also need to call this function in contexts where you're not transforming anything, but simply want to know if it can/can't be traced. I know that "can't be traced" also means "can't be transformed", but this function is really only concerned about the former, isn't it?

Reply:
> You said above that you'll also need to call this function in contexts where you're not transforming anything, but simply want to know if it can/can't be traced

Maybe there's confusion with the `no_trace` attribute? The `is_untraceable` function has only one call site.

Reply:
OK.